src/irmin-pack/unix/gc.mli

```diff
@@ -52,6 +52,8 @@ module Make (Args : Gc_args.S) : sig
       finalises. *)
 
   val cancel : t -> bool
-  val finalise_without_swap : t -> unit Lwt.t
+
+  val finalise_and_create_one_commit_store :
```

Comment:
Can you add a docstring for this? Things I think are important to highlight

- this function only works correctly if `new_store_root = new_files_path` from originating call to `v`
- the `key` should also be the same in both cases
- this function is not intended to be called directly but rather `create_one_commit_store` on the store